UPGRADE.md

```diff
@@ -132,63 +206,107 @@ Note that the upgrade will drop your existing _local_ database, so make sure to
+                    projectPath = ./.;
+
+                    # Dev tools are not needed in the release build
+                    includeDevTools = false;
+
+                    # Set optimized = true to get more optimized binaries, but slower build times
+                    optimized = false;
+                };
+                forEachSystem = nixpkgs.lib.genAttrs (import systems);
+            in
+                {
+                    # Dev shells are used for development, e.g. when running `nix develop --impure`
+                    devShells = forEachSystem (system: {
+                        default = let pkgs = nixpkgs.legacyPackages.${system}; in devenv.lib.mkShell {
+                            inherit inputs pkgs;
+                            modules = [devenvConfig];
+                        };
+                    });
+                    # Binaries for deploying IHP apps. These are used by `nix build --impure`
+                    defaultPackage = forEachSystem (system: releaseEnv nixpkgs.legacyPackages.${system});
+                };
+
+        # The following is needed to use the IHP binary cache.
+        # This binary cache provides binaries for all IHP packages and commonly used dependencies for all nixpkgs versions used by IHP.
+        nixConfig = {
+            extra-substituters = "https://digitallyinduced.cachix.org";
+            extra-trusted-public-keys = "digitallyinduced.cachix.org-1:y+wQvrnxQ+PdEsCt91rmvv39qRCYzEgGQaldK26hCKE=";
+        };
     }
     ```
-    
-    After that adjust the `default.nix` to read it's packages from the `devenv.nix`:
+
+    After that adjust the `default.nix` to read it's packages from the `flake.nix`. It's currently there for backwards compatibility reasons, but will be removed in the future:
 
     ```nix
-    # default.nix
-    let
-        ihp = ...;
-        haskellEnv = import "${ihp}/NixSupport/default.nix" {
-            ihp = ihp;
-            haskellDeps = (import ./devenv.nix { pkgs = {}; inputs = {}; config = {}; }).ihp.haskellPackages;
-            otherDeps = pkgs: (import ./devenv.nix { inherit pkgs; inputs = {}; config = {}; }).packages;
-            projectPath = ./.;
-        };
-    in
-        haskellEnv
+    # For backwards compatibility using flake.nix
+    (import
+        (
+            fetchTarball {
+                url = "https://github.com/edolstra/flake-compat/archive/12c64ca55c1014cdc1b16ed5a804aa8576601ff2.tar.gz";
+                sha256 = "0jm6nzb83wa6ai17ly9fzpqc40wg1viib8klq8lby54agpl213w5";
+            }
+        )
+    { src = ./.; }).defaultNix
     ```
 
-This means that from now on when adding new packages, you need to do it in a single file - `devenv.nix`
+This means that from now on when adding new packages, you need to do it in a single file - `flake.nix`
 
-9. **Copy settings from `Config/nix/nixpkgs-config.nix` to `devenv.nix`**
+8. **Copy settings from `Config/nix/nixpkgs-config.nix` to `flake.nix`**
```

Comment:
The section needs updating, but I don't know how

Reply:
Here's an example using `ihp.dontCheckPackages = [ "my-failing-package" ];`:

```
{
    inputs = {
        ihp.url = "github:digitallyinduced/ihp?ref=abb513016b372f9f76b6c95caed66def536a885a";
        ihp.flake = false;

        nixpkgs.url = "github:NixOS/nixpkgs?rev=a95ed9fe764c3ba2bf2d2fa223012c379cd6b32e";

        systems.url = "github:nix-systems/default";
        devenv.url = "github:cachix/devenv";
    };

    outputs = { self, nixpkgs, devenv, systems, ihp, ... } @ inputs:
        let
            devenvConfig = { pkgs, inputs, config, ... }: {
                # See full reference at https://devenv.sh/reference/options/
                # For IHP specific options, see https://ihp.digitallyinduced.com/Guide/package-management.html

                imports = [ "${inputs.ihp}/NixSupport/devenv.nix" ];
                
                ihp.enable = true;
                ihp.projectPath = ./.;

                ihp.haskellPackages = p: with p; [
                    # Haskell dependencies go here
                    p.ihp
                    cabal-install
                    base
                    wai
                    text
                    hlint
                ];

                packages = with pkgs; [
                    # Native dependencies, e.g. imagemagick
                ];

                ihp.dontCheckPackages = [ "my-failing-package" ];
            };
            
            releaseEnv = pkgs: import "${ihp}/NixSupport/default.nix" {
                ihp = ihp;
                haskellDeps = (devenvConfig inputs).ihp.haskellPackages;
                otherDeps = p: (devenvConfig inputs).packages;
                projectPath = ./.;
                includeDevTools = false;
                optimized = false;
            };
            forEachSystem = nixpkgs.lib.genAttrs (import systems);
        in
            {
                devShells = forEachSystem (system: {
                    default = let pkgs = nixpkgs.legacyPackages.${system}; in devenv.lib.mkShell {
                        inherit inputs pkgs;
                        modules = [devenvConfig];
                    };
                });
                packages = forEachSystem (system: { default = releaseEnv nixpkgs.legacyPackages.${system}; });
            };
}
```

Reply:
Where does the `doJailbreakPackages` go, under `ihp.dontCheckPackages`?

Reply:
yes below like this:
```nix
ihp.dontCheckPackages = [ "my-failing-package" ];
ihp. doJailbreakPackages = [ "some-package" ];
```

Reply:
done, added a shorter diff